### PR TITLE
Fix ride station index related bugs.

### PR DESCRIPTION
**Pull request by duncanspumpkin**
There are still a few issues. Need to make the same changes to the import code

### CODE REVIEW

File: src/openrct2/rct12/RCT12.cpp [hunk 1]
```diff
@@ -190,7 +190,11 @@ uint8_t RCT12TrackElement::GetColourScheme() const
 
 uint8_t RCT12TrackElement::GetStationIndex() const
 {
-    return (sequence & RCT12_TRACK_ELEMENT_SEQUENCE_STATION_INDEX_MASK) >> 4;
+    if (trackType == TRACK_ELEM_END_STATION || trackType == TRACK_ELEM_BEGIN_STATION || trackType == TRACK_ELEM_MIDDLE_STATION)
```
Comment (Gymnasiast) [inline src/openrct2/rct12/RCT12.cpp]: I'd still prefer a helper method for this, even if it just involves copying it from the regular tile element.

**Comment by duncanspumpkin** [comment]
I was just thinking this approach is the best for preventing duplication and makes it easier for any other person to work out the tile element quirks

**Comment by Gymnasiast** [comment]
Well, I'm ok with it. If you address the check for station elements (the line note), it's good to go AFAIC.

**Comment by Gymnasiast** [comment]
I have merged it now to prevent more corruption of people's maps.

**Pull request merged**